### DISCUSSION TITLE
Construction points and lines / (annotation points and lines)

### DISCUSSION THREAD

**Discussion by ponceta** [QGEP/QGEP#671] [Ideas]
Hi,

It would be convenient to have construction points and lines directly on the QGEP project.

These would be used to store GPS measurements and so on and would be useful for digitalization purposes.

Example :

https://github.com/qwat/qwat-data-model/blob/master/drawing/constructionpoint.sql#L8-L25

**Comment by varrieta** [comment]
I agree with that. For my databases I have added a construction_point layer but it could be useful to have it already in the project.

**Comment by ponceta** [comment]
I've made a first draft:

https://gist.github.com/ponceta/24794bd03c7dc4cba9505b8fd4da6783

Processes need to be clear before implementing any solution on this, otherwise we risk to add tables and not features. Example :

**Construction points** : construction points are artificials points to build objects.

- [ ] Document how it's meant to be used
- [ ] Create datamodel modifications
- [ ] Add layer config to the QGIS project

**Comment by max-trolliet** [comment]
The city of Lausanne developped a pluggin for construction 

https://plugins.qgis.org/plugins/VDLTools/
https://github.com/gusthiot/VDLTools

I did not try it deeply yet though. We'll try that out today and make a feedback with Kevin

**Comment by max-trolliet** [comment]
the pluggin is not (yet?) for qgis 3. I'll try it on my 2.18 at work tomorow and contact Lausanne for some input on that issue.

**Comment by max-trolliet** [comment]
Lausanne replied that the pluggin will be upgraded for QGIS 3.4. This should be schedueld for 1st trimester of 2019 (this will be synced with QWAT)

**Comment by max-trolliet** [comment]
The other pluggin available is from @3nids : Intersect it 

I personnaly think that those construction layers are really user-related. A certain user will not need the same tool as another one. Therefore, I don't think QGEP should advocate for one way or another... There are plenty of solutions for digitizing and it is up to the user or the administrator to choose which to be used.

**Comment by m-kuhn** [comment]
I assume that there will be people using QGEP which have no previous experience with QGIS. For those people it would be nice to get advice for what to use (at least one tool, potentially list other candidates).
Advanced users are not bound to this advice.

**Comment by max-trolliet** [comment]
True... 

We could add something in the documentation, advising the use of one plugin or choose to add the layers as virtual layers right away in the project... ? 

As @3nids made the Intersect it plugin for qwat, it could be interesting to promote this one. The media beeing slightly different, it seems to me that it offers more than need be, meaning everything needed for qgep already is in the Intersect it's plugin. 

@3nids Do you know if the plugin will be available for qgis 3.x soon ? Are you still the one in charge of it?

**Comment by 3nids** [comment]
There was no order yet to port it to QGIS 3.
I am just not sure if I really shall port this one.
I think the main feature missing in core is the 2 distances intersection, and there could be a simple way of adding this to existing advance digitizing tools (aka cad dock).

**Comment by ponceta** [comment]
On QGIS 3 CAD dock is a great tool, it takes a bit to get into shortcuts and workflow but it's working great. If there are any improvements to add, it should be directly corrected/pimped.

**Comment by ponceta** [comment]
Regarding survey, drawing and annotation, we have 3 different use case discussed here https://github.com/qwat/QWAT/issues/176 too.
